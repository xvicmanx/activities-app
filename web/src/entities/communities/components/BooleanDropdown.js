import React from 'react';

import { DropDown } from '../../../core/components';

const BooleanDropdown = ({ name, value, onChange }) => (
  <DropDown
    name={name}
    items={[
      {
        id: true,
        name: 'Si',
      },
      {
        id: false,
        name: 'No',
      },
    ]}
    onChange={onChange}
    placeholder="Seleccione"
    value={value}
  />
);

export default BooleanDropdown;
