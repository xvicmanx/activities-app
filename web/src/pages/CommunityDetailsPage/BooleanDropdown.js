import React, { useEffect } from 'react';
import { Heading } from 'react-bulma-components';
import { useSelector, useDispatch } from 'react-redux';

import { fetchUsers } from '../../redux/Users/UsersActions';

import {
  DropDown,
  LoadingIndicator,
} from '../../components';
import { readTokenFromCookie } from '../../redux/Users/UsersActions';

const BooleanDropdown = ({ name, value, onChange }) => (
  <DropDown
    name={name}
    items={
      [
        {
          id: true,
          name: 'Si',
        },
        {
          id: false,
          name: 'No',
        },
      ]
    }
    onChange={onChange}
    placeholder="Seleccione"
    value={value}
  />
);

export default BooleanDropdown;
