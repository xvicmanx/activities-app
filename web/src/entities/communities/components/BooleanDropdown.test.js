import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import BooleanDropdown from './BooleanDropdown';

describe('BooleanDropdown', () => {
  it('renders properly', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(
      <BooleanDropdown
        name="test-name"
        value="test-value"
        onChange={jest.fn()}
      />
    );
    expect(result).toMatchSnapshot();
  });
});
