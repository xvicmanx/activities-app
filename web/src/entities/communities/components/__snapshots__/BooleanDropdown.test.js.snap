// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooleanDropdown renders properly 1`] = `
<DropDown
  items={
    Array [
      Object {
        "id": true,
        "name": "Si",
      },
      Object {
        "id": false,
        "name": "No",
      },
    ]
  }
  name="test-name"
  onChange={[MockFunction]}
  placeholder="Seleccione"
  value="test-value"
/>
`;
